src/python/WMCore/Database/ExecuteDAO.py

```diff
@@ -0,0 +1,287 @@
+
+        self.dbi = myThread.dbi
+        self.package = package
+        self.daoModule = daoModule
+
+        # Avoid any name that starts with _
+        self.sqlRegEx = re.compile("^(?!_.*)", re.IGNORECASE)
+
+        self.daoFactory = DAOFactory(package=package,
+                                     logger=self.logger,
+                                     dbinterface=self.dbi)
+        self.logger.info("DAO Factory initialised from package: %s", self.package)
+
+        self.dao = self.daoFactory(classname=daoModule)
+        self.logger.info("DAO Module initialised as: %s", self.daoModule)
+
+    def __call__(self, *sqlArgs, dryRun=False, daoHelp=False, **sqlKwArgs):
+        """
+        __call__
+        The ExecuteDAO call method. This is the method to forward all provided
+        arguments to the execute method of the DAO and return the result from the query
+        :param dryRun:      Bool flag to indicate dryrun method
+        :param *sqlArgs:    All positional arguments to be forwarded to the DAO's execute method.
+        :param **sqlKwArgs: All named arguments to be forwarded to the DAO's execute method.
+        :return:            The result from the DAO. Depending on the DAO itself it Could be one of:
+                            * A dictionary
+                            * A list
+                            * A generator
+        """
+        if dryRun:
+            results = []
+            if daoHelp:
+                self.getHelp()
+            self.logger.info("DAO SQL queries to be executed:")
+            sqlQueries = self.getSqlQuery()
+            for sqlName, sqlStr in sqlQueries.items():
+                msg = "\n----------------------------------------------------------------------\n"
+                msg += "%s: %s"
+                msg += "\n----------------------------------------------------------------------\n"
+                self.logger.info(msg, sqlName, sqlStr)
+            self.logger.info("DAO SQL arguments provided:\n%s, %s", pformat(sqlArgs), pformat(sqlKwArgs))
+        else:
+            results = self.dao.execute(*sqlArgs, **sqlKwArgs)
+            self.logger.info("DAO Results:\n%s", pformat(results if isinstance(results, dict) else list(results)))
+        return results
+
+    def getSqlQuery(self):
+        """
+        A simple method to inspect all DAO object attributes and accumulate any sql query it finds in a simple list
+        :return: A list of all sql queries it finds in the object.
+        """
+        sqlQueries = {}
+        for attr in dir(self.dao):
+            if self.sqlRegEx.match(attr) and isinstance(getattr(self.dao, attr), (str, bytes)):
+                sqlQueries[attr] = getattr(self.dao, attr)
+        return sqlQueries
+
+    def getHelp(self):
+        """
+        A simple method to generate interactive DAO help message from the DAO source.
+        """
+        help(self.dao)
+
+
+def strToDict(dString, logger=logging.getLogger()):
```

Comment:
I guess you wanted to have `logger=None` and in the function body something like:
```
logger = logger or logging.getLogger()
```
?

Reply:
I guess this would work too. I'll change it.